Modules/Scripted/DICOMPlugins/DICOMScalarVolumePlugin.py

```diff
@@ -224,10 +230,40 @@ def examineFiles(self, files):
             for tag in subseriesTags:
                 value = slicer.dicomDatabase.fileValue(file, self.tags[tag])
                 value = value.replace(",", "_")  # remove commas so it can be used as an index
+
                 if tag not in subseriesValues:
                     subseriesValues[tag] = []
-                if not subseriesValues[tag].__contains__(value):
+
+                if tag == "imageOrientationPatient" or tag == "diffusionGradientOrientation":
+                    if value != "":
+                        directionCosine = [float(i) for i in value.split("\\")]
+                        a = math.sqrt(directionCosine[0]*directionCosine[0] + directionCosine[1]*directionCosine[1] + directionCosine[2]*directionCosine[2])
+                        for k in range(3):
+                            directionCosine[k] /= a
+                        a = math.sqrt(directionCosine[3]*directionCosine[3] + directionCosine[4]*directionCosine[4] + directionCosine[5]*directionCosine[5])
+                        for k in range(3, 6):
+                            directionCosine[k] /= a
+
+                        found = False
+                        for subseriesValue in subseriesValues[tag]:
+                            subserDirCos = [float(i) for i in subseriesValue.split("\\")]
+                            a = math.sqrt(subserDirCos[0]*subserDirCos[0] + subserDirCos[1]*subserDirCos[1] + subserDirCos[2]*subserDirCos[2])
+                            b = (directionCosine[0]*subserDirCos[0] + directionCosine[1]*subserDirCos[1] + directionCosine[2]*subserDirCos[2]) / a
```

Comment:
Here again, more descriptive names for `a` and `b` and less math noise would make this easier to understand.

Reply:
you are right, comment applied

Reply:
I don't understand why the comparison is done like this. It looks complex and maybe there is a reason for it but it is not obvious at all.

Comparing orientations looks easy but it is actually super hard. Several ITK core developers struggled with it for about a year and it took them several iterations to get it right - so that they can detect real differences but not harmless differences due to numerical imprecision. I'll try to find the code.

Reply:
I cleaned the code, see https://github.com/Slicer/Slicer/pull/7362/files#diff-ddb2464152946e0fec0e7fcd1ec6cc4aefcd6b8a9ad3ee82a136564525f0c66fR244-R272

it is a simple dot product and check if it is paralle or antiparallel (-1, +1). I took the code from the load function in the vtkITKArchetypeImageSeriesReader, i.e.  https://github.com/Slicer/Slicer/blob/1cbb44434ed36d7b7bdcb9d8a207b673571a4c0c/Libs/vtkITK/vtkITKArchetypeImageSeriesReader.h#L524-L531

Reply:
ok @lassoan @pieper changed the code to use the same behaviour of ITK numeric comparison logic (absolute differences). See:

https://github.com/Slicer/Slicer/pull/7362/files#diff-ddb2464152946e0fec0e7fcd1ec6cc4aefcd6b8a9ad3ee82a136564525f0c66fR250-R266